### DISCUSSION TITLE
Dynamically change proxy in AsyncClient

### DISCUSSION THREAD

**Discussion by ManiMozaffar** [encode/httpx#2939] [Potential Issue]
following #1376 I still think this is an issue
I'm trying to send requests from 1k ips, 10 at a time, so I need to create 1k client, this seems to be an overheard, where ALL other python request libraries allow you to change proxy on fly per request.

**Comment by karpetrosyan** [comment]
You can always do such things by accessing the private API, like this:

```python
import httpx
from httpx._utils import URLPattern

client = httpx.Client()

for i in range(10):
    client._mounts = {
        URLPattern("all://"): httpx.HTTPTransport(proxy=get_random_proxy())
    }
    client.get("https://example.com")
```

But I don't really recommend using this approach.
Instead, you can write your own client or transport that supports that.
